src/libsync/propagateremotemkdir.cpp

```diff
@@ -36,13 +38,15 @@ void PropagateRemoteMkdir::start()
     propagator()->_activeJobList.append(this);
 
     if (!_deleteExisting) {
-        return slotStartMkcolJob();
+        slotMkdir();
+        return;
     }
 
     _job = new DeleteJob(propagator()->account(),
         propagator()->_remoteFolder + _item->_file,
         this);
-    connect(_job, SIGNAL(finishedSignal()), SLOT(slotStartMkcolJob()));
+    connect(static_cast<DeleteJob*>(_job.data()), &DeleteJob::finishedSignal,
```

Comment:
From someone without profound knowledge of the architecture: why `data()`, what does it contain? Is the cast necessary?

Reply:
_job is a QPointer<AbstractNetworkJob> so we need to downcast back to the DeleteJob we just stored in there. There'd have been nicer ways of doing this but I'm aiming for the minimal changes possible for now.